Source/WebGPU/WGSL/GlobalVariableRewriter.cpp

```diff
@@ -551,7 +554,19 @@ Packing RewriteGlobalVariables::packingForType(const Type* type)
     return type->packing();
 }
 
-void RewriteGlobalVariables::collectGlobals()
+static unsigned buffersForStage(const Configuration& configuration, ShaderStage stage)
+{
+    switch (stage) {
+    case ShaderStage::Compute:
+        return configuration.maxBuffersForComputeStage;
+    case ShaderStage::Vertex:
+        return configuration.maxBuffersPlusVertexBuffersForVertexStage;
+    case ShaderStage::Fragment:
+        return configuration.maxBuffersForFragmentStage;
```

Comment:
When ever I see code like this, I alway think that perhaps an array indexed by stage would be more appropriate instead of separate variables.

Reply:
Good idea, I'm going to say it is outside the scope of this change